CIScripts/Tests/SNATTest.ps1

```diff
@@ -1,3 +1,350 @@
+        rt -c -v $Using:InternalVrf -f 1 -e $Using:ContainerNetInfo.MacAddress -n $Using:ContainerNH
+
+        # Register SNAT's left adapter ("int")
+        vif --add $Using:SNATLeftName --mac $Using:SNATLeft.MacAddress --vrf $Using:InternalVrf --type virtual --vif $Using:SNATLeftVif
+        nh --create $Using:SNATLeftNh --vrf $Using:InternalVrf --type 2 --el2 --oif $Using:SNATLeftVif
+        rt -c -v $Using:InternalVrf -f 1 -e $Using:SNATLeft.MacAddress -n $Using:SNATLeftNh
+
+        # Register SNAT's right adapter ("gw")
+        vif --add $Using:SNATRightName --mac $Using:SNATRight.MacAddress --vrf $Using:ExternalVrf --type virtual --vif $Using:SNATRightVif
+        nh --create $Using:SNATRightNh --vrf $Using:ExternalVrf --type 2 --el2 --oif $Using:SNATRightVif
+        rt -c -v $Using:ExternalVrf -f 1 -e $Using:SNATRight.MacAddress -n $Using:SNATRightNh
+
+        # Register SNAT''s additional adapter ("veth")
+        vif --add $Using:SNATVeth.Name --mac $Using:SNATVeth.MacAddress --vrf $Using:ExternalVrf --type virtual --vif $Using:SNATVeth.vif
+        nh --create $Using:SNATVeth.nh --vrf $Using:ExternalVrf --type 2 --el2 --oif $Using:SNATVeth.vif
+        rt -c -v $Using:ExternalVrf -f 1 -e $Using:SNATVeth.MacAddress -n $Using:SNATVeth.nh
+
+        # Broadcast NH (internal network)
+        nh --create $Using:BroadcastInternalNh --vrf $Using:InternalVrf --type 6 --cen --cni $Using:ContainerNH --cni $Using:SNATLeftNh
+        rt -c -v $Using:InternalVrf -f 1 -e ff:ff:ff:ff:ff:ff -n $Using:BroadcastInternalNh
+
+        # Broadcast NH (external network)
+        nh --create $Using:BroadcastExternalNh --vrf $Using:ExternalVrf --type 6 --cen --cni $Using:SNATRightNh --cni $Using:SNATVeth.nh
+        rt -c -v $Using:ExternalVrf -f 1 -e ff:ff:ff:ff:ff:ff -n $Using:BroadcastExternalNh
+    }
+    Write-Host "Configure vRouter... DONE"
+
+    Write-Host "Configure endhost..."
+    $PhysicalMac = $PhysicalAdapter.MacAddress
+    $GatewayIP = $SNATConfiguration.GatewayIP
+    $EndhostPassword = $SNATConfiguration.EndhostPassword
+    $EndhostSecurePassword = ConvertTo-SecureString $SNATConfiguration.EndhostPassword -AsPlainText -Force
+    $EndhostCredentials = New-Object System.Management.Automation.PSCredential($SNATConfiguration.EndhostUsername, $EndhostSecurePassword)
+    Invoke-Command -Session $Session -ScriptBlock {
+        $EndhostSession = New-SSHSession -IPAddress $Using:SNATConfiguration.EndhostIP -Credential $Using:EndhostCredentials -AcceptKey
+        Invoke-SSHCommand -SessionId $EndhostSession.SessionId `
+            -Command "echo $Using:EndhostPassword | sudo -S arp -i eth0 -s $Using:GatewayIP $Using:PhysicalMac" | Out-Null
+    }
+    Write-Host "Configure endhost... DONE"
+
+    $Res = Invoke-Command -Session $Session -ScriptBlock {
+        docker exec $Using:ContainerID ping $Using:SNATConfiguration.EndhostIP | Write-Host
```

Comment:
I think to keep it consistent, this should be moved to `Test-CanPingEndhostFromContainer` function